stdlib/dynarray.ml

```diff
@@ -422,29 +460,28 @@ let reset a =
    against.)
 *)
 
-(* [add_last_if_room a elem] only writes the slot if there is room, and
+(* [add_last_if_room a v] only writes the value if there is room, and
    returns [false] otherwise. *)
-let[@inline] add_last_if_room a elem =
-  let {arr; length} = a in
+let[@inline] add_last_if_room a v =
+  let {arr; length; _} = a in
   (* we know [0 <= length] *)
   if length >= Array.length arr then false
   else begin
     (* we know [0 <= length < Array.length arr] *)
     a.length <- length + 1;
-    Array.unsafe_set arr length elem;
+    Array.unsafe_set arr length v;
     true
   end
 
 let add_last a x =
-  let elem = Elem {v = x} in
-  if add_last_if_room a elem then ()
+  if add_last_if_room a x then ()
   else begin
     (* slow path *)
-    let rec grow_and_add a elem =
+    let rec grow_and_add a x =
       ensure_extra_capacity a 1;
-      if not (add_last_if_room a elem)
-      then grow_and_add a elem
-    in grow_and_add a elem
+      if not (add_last_if_room a x)
+      then grow_and_add a x
+    in grow_and_add a x
   end
 
 let rec append_list a li =
```

Comment:
I don't remember, but I think that this made a noticeable difference in benchmarks. (If that was the reason, it would have been nice to write it down in a comment.)